src/briefcase/integrations/android_sdk.py

```diff
@@ -1412,6 +1410,8 @@ def start_emulator(
                 self.tools.logger.info(general_error_msg)
 
             raise
+        finally:
+            emulator_streamer.stop_flag.set()
```

Comment:
I'd be inclined to wrap this as a `streamer.stop()` method  (or similar), rather than reaching into the internal implementation so if we ever have more complex "kill the streamer" logic, we don't miss anything that is required.

Reply:
That makes sense; although, I'm inclined to call it `streamer.request_stop()` because `readline()` is blocking and as such, we can't guarantee setting this flag will actually ever cause the thread to exit (....you know, if its stuck in `readline()`).

Reply:
Sure the exact name doesn't really concern me, it's more the conceptual wrapping of the behavior, rather than reaching into internals. Also `stop()` could easily be confused for a method on the core thread API (it doesn't AFAICT, but `stop()` would be an obvious conceptual match for `run()`... but it won't behave that way).

Reply:
RE: stopping threads

I always loved this blog post from raymond chen....although, he seemingly removed it...

https://web.archive.org/web/20190108101649/https://blogs.msdn.microsoft.com/oldnewthing/20150814-00/?p=91811